### ISSUE TITLE
Channel not found when importing from external drive

### ISSUE THREAD

**Issue by lyw07** [learningequality/kolibri#5228]
### Observed behavior

On my OSX machine, when importing a channel from external drive, after click `SELECT` button, it shows an error saying that `Channel not found`. However, if I refresh the page, I will be able to see the channel detail page. 
Note that this does not happen on Ubuntu.

### Expected behavior

Users should be able to import channels successfully from external drives.

### User-facing consequences

Users will get confused if they see the error without refreshing the page.

### Errors and logs

<img width="474" alt="screen shot 2019-02-28 at 9 48 04 pm" src="https://user-images.githubusercontent.com/19424916/53619198-54355480-3ba3-11e9-92fe-bc671c01cca3.png">

<img width="266" alt="screen shot 2019-02-28 at 9 47 46 pm" src="https://user-images.githubusercontent.com/19424916/53619195-513a6400-3ba3-11e9-820a-b97d522780c7.png">

<img width="474" alt="screen shot 2019-02-28 at 9 48 19 pm" src="https://user-images.githubusercontent.com/19424916/53619204-5697ae80-3ba3-11e9-9ed6-1e867950082d.png">
<img width="291" alt="screen shot 2019-02-28 at 9 48 32 pm" src="https://user-images.githubusercontent.com/19424916/53619207-59929f00-3ba3-11e9-8d7d-a29f46676c32.png">
<img width="640" alt="screen shot 2019-02-28 at 9 48 54 pm" src="https://user-images.githubusercontent.com/19424916/53619211-5bf4f900-3ba3-11e9-90ce-dd4b12d5b2da.png">

### Steps to reproduce

1. start kolibri
2. import some content from studio
3. export the content to external drive
4. delete the content on kolibri
5. try to import the content from external drive
6. click `SELECT` button in the local import channel page

### Context

Kolibri version: 0.12.0b11
Operating system: OSX 10.14.2
Browser: Chrome (Version 72.0.3626.109)

**Comment by jonboiser** [comment]
@lyw07 On Mac, do you see the "generating channel listing" progress bar after refreshing? How about when you see the "Channel not found" error?

The DeviceChannel endpoint draws on ChannelMetadata resource, so I'm guessing that you are being redirected to the channel detail page before the ChannelMetadata from the external drive is finished being downloaded and registered on Kolibri.

**Comment by lyw07** [comment]
Hi @jonboiser, I recorded what I did, thinking that it might be easier for you to see what happened. Thanks!
![local_import_error](https://user-images.githubusercontent.com/19424916/53668123-d10a1200-3c27-11e9-8b0b-eb816df2a260.gif)

**Comment by jonboiser** [comment]
I'm able to reproduce this too. I'm noticing that the Task object looks unusual because it's "Completed" but has percentage 0. I'm wondering if there is code that is relying on percentage being 1. The tasks is also not causing a progress bar too appear on the top, so it can be cancelled and cleared out.

And to answer my own question, the "generating channel listing" progress bar does not appear in this scenario.

```
{status: "COMPLETED", exception: "None", cancellable: true, percentage: 0, traceback: "",…}
cancellable: true
exception: "None"
id: "91d9f0e894ab46ce94981e0829b52e77"
percentage: 0
started_by: "1979b2a7deb2b6637fbae39096e671e2"
status: "COMPLETED"
traceback: ""
type: "DISKCHANNELIMPORT"
```

**Comment by lyw07** [comment]
I think the percentage for tasks never gets to 1 actually. One thing I notice about the `Channel not found` error is that the call to `http://localhost:8080/api/content/contentnode_granular/` happened when the `DISKCHANNELIMPORT` task was still in progress or queue, before it was completed.

**Comment by jonboiser** [comment]
> happened when the DISKCHANNELIMPORT task was still in progress or queue, before it was completed.

~I haven't seen this. I've only seen the one Task that is COMPLETED. I don't see the usual tasks that precede it. For me, the Tasks API response is `[]`, then the next second is it is the COMPLETED task.~

Things are definitely happening out of order. I'm seeing the call to DeviceChannel before there are even tasks coming back from the Tasks endpoint.

The expected behavior is that you get tasks with status not `COMPLETED` (queued, wscheduled, running) and only when `percentage` is 1, you get to `COMPLETED`. This means the channel metadata DB has been downloaded from the source (e.g. the external drive) and added to the ChannelMetadata and ContentNode tables.

The reason we're getting 404 on Macs is that (I think) the Tasks' status is immediately going to `COMPLETED`, which causes the import/export wizard to go to the Channel Detail page (select contents page), before all of that data is available.

**Comment by lyw07** [comment]
one thing weird is that now it works on my mac 😂

**Comment by jonboiser** [comment]
Uh oh, this might be a probabilistic bug. I'll see if it's working for me too.

**Comment by radinamatic** [comment]
TIL the cause of so many of my rants actually has a (rather appropriate) name: _probabilistic bug_... :roll_eyes:

**Comment by jonboiser** [comment]
I think part of the problem is that the DISKCHANNELIMPORT type tasks weren't causing the vuex task list to update. I added a fix for this in #5233, which seems to mitigate the problem in this issue, although there are still some weird behaviors, like the premature DeviceChannel endpoint call.

**Comment by jonboiser** [comment]
This should be fixed by 4244766 in #5233.
